diff -up p11-kit-0.25.0/meson.build.omv~ p11-kit-0.25.0/meson.build
--- p11-kit-0.25.0/meson.build.omv~	2023-07-10 12:59:43.405779798 +0100
+++ p11-kit-0.25.0/meson.build	2023-07-10 13:07:32.526541068 +0100
@@ -310,8 +310,13 @@ int main (void)
     return strerror_r (EINVAL, buf, 32);
 }
 '''
-  strerror_r_check = cc.run(strerror_r_code, name : 'strerror_r check')
-  if strerror_r_check.returncode() == 0
+  # Upstream does cc.run here; that breaks crosscompiling.
+  # Fortunately the code also triggers a -Wint-conversion, so
+  # by turning that into an error, just checking if it compiles
+  # is sufficient. (Of course this breaks crappy compilers. Less
+  # bad than breaking crosscompiles.)
+  add_project_arguments(['-Werror=int-conversion'], language: 'c')
+  if cc.compiles(strerror_r_code, name : 'strerror_r check')
     conf.set('HAVE_XSI_STRERROR_R', 1)
   else
     conf.set('HAVE_GNU_STRERROR_R', 1)
